dm/master/server.go

```diff
@@ -1332,6 +1346,62 @@ func (s *Server) generateSubTask(ctx context.Context, task string) (*config.Task
 	return cfg, stCfgs, nil
 }
 
+func (s *Server) removeMetaData(ctx context.Context, cfg *config.TaskConfig) error {
+	toDB := *cfg.TargetDB
+	toDB.Adjust()
+	if len(toDB.Password) > 0 {
+		pswdTo, err := utils.Decrypt(toDB.Password)
+		if err != nil {
+			return err
+		}
+		toDB.Password = pswdTo
+	}
+
+	sqls := make([]string, 0, 3)
+	sqls = append(sqls, fmt.Sprintf("CREATE SCHEMA IF NOT EXISTS `%s`", cfg.MetaSchema))
+	// clear loader and syncer checkpoints
+	sqls = append(sqls, fmt.Sprintf("DROP TABLE IF EXISTS `%s`.`%s`",
+		cfg.MetaSchema, cfg.Name+"_loader_checkpoint"))
+	sqls = append(sqls, fmt.Sprintf("DROP TABLE IF EXISTS `%s`.`%s`",
+		cfg.MetaSchema, cfg.Name+"_syncer_checkpoint"))
+
+	// clear etcd data
+	if cfg.ShardMode == config.ShardPessimistic {
+		err := s.pessimist.RemoveMetaData(cfg.Name)
+		if err != nil {
+			return err
+		}
+		// clear shard meta data for pessimistic
+		sqls = append(sqls, fmt.Sprintf("DROP TABLE IF EXISTS `%s`.`%s`",
+			cfg.MetaSchema, fmt.Sprintf(shardmeta.MetaTableFormat, cfg.Name)))
+	} else if cfg.ShardMode == config.ShardOptimistic {
+		err := s.optimist.RemoveMetaData(cfg.Name)
+		if err != nil {
+			return err
+		}
+	}
+
+	// clear online ddl schema
+	if cfg.OnlineDDLScheme != "" {
```

Comment:
How about always trying to clear online DDL schema (to handle the case where `online-ddl-scheme` exists before but removed now)?

Reply:
Now we will delete all